File: spec/javascript/test_process_custom.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ProcessCustom', 'src/process_rotate.bin', function(r) {
    assert.equal(r.buf1.toString(), new Uint8Array([0x10, 0xb3, 0x94, 0x94, 0xf4]).toString());
    assert.equal(r.buf2.toString(), new Uint8Array([0x5f, 0xba, 0x7b, 0x93, 0x63, 0x23, 0x5f]).toString());
    assert.equal(r.buf3.toString(), new Uint8Array([0x29, 0x33, 0xb1, 0x38, 0xb1]).toString());
});

File: spec/cpp_stl/test_process_custom.cpp
#include <boost/test/unit_test.hpp>

#include <process_custom.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_process_custom) {
    std::ifstream ifs("src/process_rotate.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    process_custom_t* r = new process_custom_t(&ks);

    BOOST_CHECK_EQUAL(r->buf1(), std::string("\x10\xb3\x94\x94\xf4", 5));
    BOOST_CHECK_EQUAL(r->buf2(), std::string("\x5f\xba\x7b\x93\x63\x23\x5f", 7));
    BOOST_CHECK_EQUAL(r->buf3(), std::string("\x29\x33\xb1\x38\xb1", 5));

    delete r;
}
